### DISCUSSION TITLE
Avoid object detection on specific areas

### DISCUSSION THREAD

**Discussion by JanickGers85** [blakeblackshear/frigate#7693] [Ask A Question]
Hello there,
being struggling for a few time, so I'd like to ask.
I have some false positive on my camera so I created a zone to exclude persons events from the yellow area, but objects are still detected even if they do not produce any event.
How to avoid it on specific areas?

Excluded area are is in yellow
![IMG_20230905_115217_41](https://github.com/blakeblackshear/frigate/assets/21289907/da9b6528-2143-42a8-9a03-a92bd4f6cb44)

**Comment by Baael** [comment]
No idea, but the same here

motion and object mask:
<img width="573" alt="Zrzut ekranu 2023-09-04 114129" src="https://github.com/blakeblackshear/frigate/assets/38595/cd2203ed-1112-44d7-b2fc-264852141c41">

and this event triggers detection:
<img width="480" alt="Zrzut ekranu 2023-09-04 114211" src="https://github.com/blakeblackshear/frigate/assets/38595/612c9f3c-56e6-4af9-89e2-11ae7f9818a7">

**Reply by NickM-27** [reply]
This type of large motion mask is discouraged in the docs, and it won't stop objects from being detected.

**Comment by NickM-27** [comment]
There is no way to do this because that's not how it works. Frigate is designed to track objects everywhere but only alert / create an event when entering a zone. If frigate didn't track objects outside the zone it wouldn't have enough time to confirm it is a true positive object once it enters the zone.

**Reply by Baael** [reply]
yeah I forgot to mention, the truck on previous post triggers detection alert on yellow zone, I added masks trying to fix it 

<img width="350" alt="Zrzut ekranu 2023-09-05 122328" src="https://github.com/blakeblackshear/frigate/assets/38595/ed2766b1-1e06-4dbd-9444-409703ab2306">

**Reply by JanickGers85** [reply]
@NickM-27 Thanks for the clear explanation.
And there are no workaround for this?
Like changing the object detection at zone level? In this case I would point Frigate to detect an object that likely won't be detected, like a bycicle in my case 😏

Another weird behaviour is that when a person and so an event is detected in the other zone, in the event list can appear a snapshot from the excluded zone: this maybe because the false positive in the excluded zone has a bigger confidence than the one detected in the included zone.
Is this an expected behaviour or a bug?

**Reply by NickM-27** [reply]
@Baael that's because an objects presence in a zone is based on the bottom center of its bounding box

**Reply by NickM-27** [reply]
@JanickGers85 you can set types of objects for each zone https://docs.frigate.video/configuration/zones#restricting-zones-to-specific-objects

I don't have enough info to understand what you mean or how your config is behaving
